src/dioptra/restapi/v1/groups/schema.py

```diff
@@ -36,3 +37,121 @@ class GroupRefSchema(Schema):
+        ),
+    )
+    shareWrite = fields.Boolean(
+        attribute="share_write",
+        metadata=dict(
+            description="Permission for member to share read+write Group Resources."
+        ),
+    )
+
+
+class MemberSchema(MemberMutableFieldsSchema):
+    """The fields schema of a Group Member."""
+
+    user = fields.Nested(
+        UserRefSchema,
+        attribute="user",
+        metadata=dict(description="User that is a member of the Group resource."),
+    )
+    group = fields.Nested(
+        GroupRefSchema,
+        attribute="group",
+        metadata=dict(description="The Group Resource of which the User is a member."),
+    )
+
+
+class ManagerSchema(MemberSchema):
+    """The fields schema of a Group Manager."""
+
+    owner = fields.Boolean(
+        attribute="owner",
+        metadata=dict(description="Flag for if the Manager is a Group owner."),
+    )
+    admin = fields.Boolean(
+        attribute="admin",
+        metadata=dict(description="Flag for if the Manager is a Group admin."),
+    )
+
+
+class GroupMutableFieldsSchema(Schema):
+    """The fields schema for the mutable data by Members in a Group resource."""
+
+    members = fields.Nested(
+        MemberSchema,
+        attribute="members",
+        metadata=dict(description="A list of Members in a Group."),
+        many=True,
+        dump_only=True,
+    )
+    name = fields.String(
+        attribute="name", metadata=dict(description="Name of the Group resource.")
+    )
+    managers = fields.Nested(
+        ManagerSchema,
+        attribute="managers",
+        metadata=dict(description="A list of Managers in a Group."),
+        many=True,
+        dump_only=True,
+    )
+
+
+class GroupSchema(GroupMutableFieldsSchema):  # type: ignore
+    """The schema for the data stored in a Group resource."""
+
+    id = fields.Integer(
+        attribute="id",
+        metadata=dict(description="ID for the Group resource."),
+    )
+    user = fields.Nested(
+        UserRefSchema,
+        attribute="user",
+        metadata=dict(description="User that created the Group resource."),
+        dump_only=True,
+    )
+    createdOn = fields.DateTime(
+        attribute="created_on",
+        metadata=dict(description="Timestamp when the Group resource was created."),
+        dump_only=True,
+    )
+    lastModifiedOn = fields.DateTime(
+        attribute="last_modified_on",
+        metadata=dict(
+            description="Timestamp when the Group resource was last modified."
+        ),
+        dump_only=True,
+    )
+    resources = fields.Url(
+        attribute="resources",
+        metadata=dict(
+            description="A url pointing to a list of Resourcess associated with the Group."
+        ),
+        relative=True,
+        dump_only=True,
+    )
+    tags = fields.Url(
+        attribute="tags",
+        metadata=dict(
+            description="A url pinting to a list of Tags associated with the Group."
```

Comment:
small typo in the description `"A url pinting..."` should read `"A url pointing..."`